src/pages/Fox/hooks/useFoxWifHatClaimed.tsx

```diff
@@ -0,0 +1,64 @@
+import { FOX_WIF_HAT_MERKLE_DISTRIBUTOR_ABI, viemClientByNetworkId } from '@shapeshiftoss/contracts'
+import { skipToken, useQuery } from '@tanstack/react-query'
+import { useMemo } from 'react'
+import { getContract } from 'viem'
+import { base } from 'viem/chains'
+
+import { FOX_WIF_HAT_MERKLE_DISTRIBUTOR_CONTRACT } from '../constant'
+
+const client = viemClientByNetworkId[base.id]
+
+type FoxWifHatClaimedQueryKey = [
+  'foxWifHatClaimed',
+  {
+    index: bigint | undefined
+  },
+]
+
+type UseFoxWifHatClaimedQueryProps = {
+  index: bigint | undefined
+}
+
+export const getFoxWifHatClaimedQueryKey = ({
+  index,
+}: UseFoxWifHatClaimedQueryProps): FoxWifHatClaimedQueryKey => [
+  'foxWifHatClaimed',
+  {
+    index,
+  },
+]
+
+export const getFoxWifHatClaimedQueryFn = ({ index }: UseFoxWifHatClaimedQueryProps) => {
+  if (!index) return skipToken
+
+  return async () => {
+    try {
+      return await getContract({
+        address: FOX_WIF_HAT_MERKLE_DISTRIBUTOR_CONTRACT,
+        abi: FOX_WIF_HAT_MERKLE_DISTRIBUTOR_ABI,
+        client,
+      }).read.isClaimed([index])
+    } catch (err) {
+      console.error(err)
+      return false
+    }
+  }
+}
+
+export const useFoxWifHatClaimedQuery = ({ index }: UseFoxWifHatClaimedQueryProps) => {
+  const queryKey: FoxWifHatClaimedQueryKey = useMemo(() => {
+    return getFoxWifHatClaimedQueryKey({ index })
+  }, [index])
+
+  const queryFn = useMemo(() => {
+    return getFoxWifHatClaimedQueryFn({ index })
+  }, [index])
+
+  const query = useQuery({
+    queryKey,
+    queryFn,
+    enabled: !!index,
```

Comment:
blocking: index 0 claimooor will never have this query enabled:

```shell
❯ node
Welcome to Node.js v18.20.4.
Type ".help" for more information.
> !!0n
false
```

```suggestion
    enabled: index !== undefined,
```

Reply:
Very nice catch! One user would have been recked 😆 

Fixed in [7e9a198](https://github.com/shapeshift/web/pull/8611/commits/7e9a198a552c9ffbe24675fdb83a91d1875e342b)